### ISSUE TITLE
occlum/occlum:0.28.1-ubuntu20.04 image is too large (6.22GB)

### ISSUE THREAD

**Issue by qiyuangong** [occlum/occlum#1007]
```bash
occlum/occlum                         0.28.1-ubuntu20.04   68c7de248fae   3 weeks ago     6.22GB
```
6.22 GB is too large for external developer to pull from internet. :)

**Comment by qzheng527** [comment]
I did a quick anlalysis by the tool [`dive`](https://github.com/wagoodman/dive) which shows the image efficiency score is 99%.
It means there is not much space to shrink the size by docker skills.
Current occlum docker image includes too many toolchains which users may not need.
The top size contribution parts are:

- dependent packages installed by `apt install`, ~1.8GB
- rust toolchain installed to compile occlum, ~1.1GB
- Occlum golang toolchain, ~800MB.
- Occlum jvm toolchains (three different jvms included), ~1GB

Therefore, to users, alternative ways could be used to seek smaller size occlum image.

1. Occlum team to provide basic occlum dev docker image, perhaps only musl-libc and glibc toolchain included.
2. Users use `apt install` way to install occlum debian packages in ubuntu 20.04 based docker image, please refer to https://github.com/occlum/occlum/tree/master/tools/installer#how-to-use
But currently only musl-gcc and golang toolchains debian packages are ready. (glibc toolchain is going to be ready).
3. If users just need `occlum run`, please refer to [`multistage deployment demo`] (https://github.com/occlum/occlum/tree/master/demos/deployment) or [`dockerfile`](https://github.com/occlum/occlum/blob/master/example/container/Dockerfile_occlum_instance.ubuntu20.04)  for reference.

**Comment by guzongmin** [comment]
The Occlum docker image is a development environment for Occlum developers. So I think the easy way for PPML developers is to create a PPML development environment base on the Occlum runtime deb packages. The size could be less than 500MB.

**Comment by qiyuangong** [comment]
> The Occlum docker image is a development environment for Occlum developers. So I think the easy way for PPML developers is to create a PPML development environment base on the Occlum runtime deb packages. The size could be less than 500MB.

We used install package a few month ago. That will greatly reduce image size. However, after occlum remove `init/build` from that package, we switched back to occlum image. 

We will try to switch to install package again. But still need to wait for @qzheng527 make glibc available.

**Comment by qiyuangong** [comment]
Issue addressed by #5989.
